File: argentum-django/api/tests/data/bonus_transactions.py
from decimal import Decimal

from django.utils.dateparse import parse_datetime

from api.models.bonus_transaction import BonusTransaction
from api.tests.data.guests import TestGuests
from api.tests.utils.test_objects import TestObjects


class TestBonusTransactions(TestObjects):
    MODEL = BonusTransaction

    BTX1 = MODEL(
        id=1,
        time=parse_datetime('2019-12-31T22:01:00Z'),
        guest=TestGuests.ROBY,
        value=Decimal('2.50'),
        description='staff bonus',
        pending=False
    )

    BTX2 = MODEL(
        id=2,
        time=parse_datetime('2019-12-31T22:02:30Z'),
        guest=TestGuests.SHEELAH,
        value=Decimal('2.00'),
        description='guest bonus',
        pending=False
    )

    BTX3 = MODEL(
        id=3,
        time=parse_datetime('2019-12-31T22:03:00Z'),
        guest=TestGuests.ROBY,
        value=Decimal('3.00'),
        description='more staff bonus',
        pending=True
    )

    ALL = [BTX1, BTX2, BTX3]

    # Models below are not stored in the DB, but rather used for POST deserialization testing.

    BTX4 = MODEL(
        id=4,
        guest=TestGuests.ROBY,
        value=Decimal('4.00'),
        description='staff bonus',
        pending=True
    )

File: argentum-ui/src/app/common/rest-service/test-data/bonus-transactions.ts
import {Guests} from './guests';
import {BonusTransaction} from '../../model/bonus-transaction';

export namespace BonusTransactions {
  export const BTX1 = new BonusTransaction(
    21010,
    new Date('2019-12-31T22:01:00Z'),
    Guests.ROBY,
    undefined,
    2.50,
    'default',
    false
  );

  export const BTX2 = new BonusTransaction(
    21020,
    new Date('2019-12-31T22:02:30Z'),
    Guests.SHEELAH,
    undefined,
    3.00,
    'default',
    false
  );

  export const BTX3 = new BonusTransaction(
    21030,
    new Date('2019-12-31T22:03:00Z'),
    Guests.ROBY,
    undefined,
    3.00,
    'default',
    true
  );

  export const ALL = [BTX1, BTX2, BTX3];

  export const BTX1_BY_CARD_REFERENCE = new BonusTransaction(
    21010,
    new Date('2019-12-31T22:01:00Z'),
    undefined,
    undefined,
    2.50,
    'default',
    false
  );

  export const BTX3_BY_CARD_REFERENCE = new BonusTransaction(
    21030,
    new Date('2019-12-31T22:03:00Z'),
    undefined,
    undefined,
    3.00,
    'default',
    true
  );

  export const BTX4_REFERENCE = new BonusTransaction(
    21040,
    new Date('2019-12-31T22:04:20Z'),
    undefined,
    undefined,
    4.00,
    'default',
    true
  );

  export const BTX3_PATCHED_RESPONSE = new BonusTransaction(
    21031,
    new Date('2019-12-31T22:03:05Z'),
    undefined,
    undefined,
    3.00,
    'default',
    false
  );
}
